File: Kirameku-backend/admin/src/api/project.ts
import { http } from "@/utils/http";

export type ProjectItem = {
  id: number;
  name: string;
  slug: string;
  description: string;
  long_description: string;
  cover_image: string;
  tech_stack: string[];
  link_github: string;
  link_gitee: string;
  link_live: string;
  link_docs: string;
  project_type: "own" | "favorite";
  status: string;
  status_label: string;
  is_featured: boolean;
  sort: number;
  created_at: string;
  updated_at: string;
};

export type ProjectMetadata = Pick<
  ProjectItem,
  | "name"
  | "slug"
  | "description"
  | "long_description"
  | "cover_image"
  | "tech_stack"
  | "link_github"
  | "link_gitee"
  | "link_live"
  | "link_docs"
>;

/** 获取项目列表 */
export const getProjects = () => {
  return http.request<ProjectItem[]>("get", "/api/projects");
};

/** 从项目托管平台读取收藏项目信息 */
export const getProjectMetadata = (sourceUrl: string) => {
  return http.request<ProjectMetadata>(
    "post",
    "/api/projects/metadata",
    { data: { source_url: sourceUrl } },
    { timeout: 30000 }
  );
};

/** 创建项目 */
export const createProject = (data: Partial<ProjectItem>) => {
  return http.request<ProjectItem>("post", "/api/projects", { data });
};

/** 更新项目 */
export const updateProject = (
  projectId: number,
  data: Partial<ProjectItem>
) => {
  return http.request<ProjectItem>("put", `/api/projects/${projectId}`, {
    data
  });
};

/** 删除项目 */
export const deleteProject = (projectId: number) => {
  return http.request<{ ok: boolean }>("delete", `/api/projects/${projectId}`);
};

File: Kirameku-backend/app/schemas/project.py
from datetime import datetime
from typing import Literal
from pydantic import BaseModel

ProjectType = Literal["own", "favorite"]


class ProjectCreate(BaseModel):
    name: str
    slug: str
    description: str = ""
    long_description: str = ""
    cover_image: str = ""
    tech_stack: list[str] = []
    link_github: str = ""
    link_gitee: str = ""
    link_live: str = ""
    link_docs: str = ""
    project_type: ProjectType = "own"
    status: str = "developing"
    status_label: str = ""
    is_featured: bool = False
    sort: int = 0


class ProjectUpdate(BaseModel):
    name: str | None = None
    slug: str | None = None
    description: str | None = None
    long_description: str | None = None
    cover_image: str | None = None
    tech_stack: list[str] | None = None
    link_github: str | None = None
    link_gitee: str | None = None
    link_live: str | None = None
    link_docs: str | None = None
    project_type: ProjectType | None = None
    status: str | None = None
    status_label: str | None = None
    is_featured: bool | None = None
    sort: int | None = None


class ProjectMetadataRequest(BaseModel):
    source_url: str


class ProjectMetadataOut(BaseModel):
    name: str
    slug: str
    description: str
    long_description: str
    cover_image: str
    tech_stack: list[str]
    link_github: str
    link_gitee: str
    link_live: str
    link_docs: str


class ProjectOut(BaseModel):
    id: int
    name: str
    slug: str
    description: str
    long_description: str
    cover_image: str
    tech_stack: list[str] = []
    link_github: str
    link_gitee: str
    link_live: str
    link_docs: str
    project_type: ProjectType
    status: str
    status_label: str
    is_featured: bool
    sort: int
    created_at: datetime
